src/clostack/signature.clj

```diff
@@ -4,14 +4,37 @@
            javax.crypto.Mac
            java.util.Base64))
 
+
+(def byte-array?
+  (partial instance? (Class/forName "[B")))
+
+
+(defn to-bytes
+  "Tries to coerce an input to a byte array. Ignores nil.
+  Throws an error when gives up."
+  [input]
+  (cond
+
+    (string? input)
+    (.getBytes ^String input)
+
+    (byte-array? input)
+    input
+
+    (some? input)
```

Comment:
I would use `:else` here to throw on anything that is not byte-coercable from our point of view.